File: domain/src/main/java/com/arturogutierrez/openticator/domain/issuer/repository/IssuerRepositoryImpl.java
package com.arturogutierrez.openticator.domain.issuer.repository;

import com.arturogutierrez.openticator.domain.issuer.model.Issuer;
import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;
import javax.inject.Inject;
import rx.Observable;

public class IssuerRepositoryImpl implements IssuerRepository {

  @Inject
  public IssuerRepositoryImpl() {

  }

  @Override
  public Observable<List<Issuer>> getIssuers() {
    List<Issuer> allIssuers = new ArrayList<>(Arrays.asList(Issuer.values()));
    allIssuers.remove(Issuer.UNKNOWN);
    return Observable.just(allIssuers);
  }
}

File: domain/src/main/java/com/arturogutierrez/openticator/domain/issuer/repository/IssuerRepositoryImpl.kt
package com.arturogutierrez.openticator.domain.issuer.repository

import com.arturogutierrez.openticator.domain.issuer.model.Issuer
import rx.Observable
import javax.inject.Inject

class IssuerRepositoryImpl @Inject constructor() : IssuerRepository {

    override val issuers: Observable<List<Issuer>>
        get() {
            val allIssuers = Issuer.values().asList().filter { it != Issuer.UNKNOWN }
            return Observable.just<List<Issuer>>(allIssuers)
        }
}
